src/compileToIntermediateTree.test.js

```diff
@@ -0,0 +1,260 @@
+# header1
+## header2
+# header3
+## header2
+  `);
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile ordered list', () => {
+  const compiled = compileToIntermediateTree(`
+1. foo
+2. bar
+  `);
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile list', () => {
+  const compiled = compileToIntermediateTree(`
+- foo
+- bar
+  `);
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile tables', () => {
+  const compiled = compileToIntermediateTree(`
+| Tables        | Are           | Cool  |
+| ------------- |:-------------:| -----:|
+| col 3 is      | right-aligned | $1600 |
+| col 2 is      | centered      |   $12 |
+| zebra stripes | are neat      |    $1 |
+  `);
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile codespans', () => {
+  const compiled = compileToIntermediateTree('install with `$ npm install`');
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile image', () => {
+  const compiled = compileToIntermediateTree('![test](http://some.com/image.png)');
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile html', () => {
+  const compiled = compileToIntermediateTree('<div>hello</div>');
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile multiple html', () => {
+  const compiled = compileToIntermediateTree(
+    '<div>hello</div>\n<strong>there</strong>\n<em>world</em>'
+  );
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile components using H and marksy language', () => {
+  const compiled = compileToIntermediateTree('```marksy\nh(Test)\n```');
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to compile nested lists', () => {
+  const compiled = compileToIntermediateTree(`
+- Colors
+    - Red
+    - Blue
+- Shape
+  - Triangle
+  - Rectangle
+  `);
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should be able to combine in compilation', () => {
+  const compiled = compileToIntermediateTree(`
+# hey
+
+- foo
+
+-bar
+  `);
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should produce TOC', () => {
+  const compiled = compileToIntermediateTree(`
+# foo
+
+## bar
+
+### baz
+  `);
+
+  expect(JSON.stringify(compiled.toc, null, 2)).toMatchSnapshot();
+});
+
+it('should produce custom tags', () => {
+  const compiled = compileToIntermediateTree(`
+# foo
+  `);
+
+  expect(compiled).toMatchSnapshot();
+});
+
+it('should work with Preact', () => {
+  const compiled = compileToIntermediateTree(`
+# foo
+  `);
+
+  expect(preactRenderToString(Preact.h('div', null, compiled.tree))).toMatchSnapshot();
+});
+
+it('should work with Inferno', () => {
+  const compiled = compileToIntermediateTree(`
+# foo
+  `);
+
+  expect(infernoRenderToString(infernoCreateElement('div', null, compiled.tree))).toMatchSnapshot();
+});
+
+it('should allow injecting context to elements', () => {
```

Comment:
What does this test case intend to test?

Reply:
context injection